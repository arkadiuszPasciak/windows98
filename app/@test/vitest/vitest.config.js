import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
