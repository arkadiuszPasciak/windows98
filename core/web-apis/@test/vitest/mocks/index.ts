export * from "./canvas-api.mock"
export * from "./media-devices-api.mock"
export * from "./screen-api.mock"
export * from "./video-api.mock"
