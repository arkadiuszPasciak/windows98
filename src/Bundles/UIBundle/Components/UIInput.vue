<template>
  <div class="UIInput" :class="`label-position-${labelPosition}`">
    <label class="label" :for="id">{{ labelName }}</label>
    <input
      :id="id"
      class="input"
      :type="type"
      :disabled="disabled"
      :readonly="readonly"
      :value="modelValue"
      @input="
        $emit('update:modelValue', ($event.target as HTMLInputElement).value)
      "
    />
  </div>
</template>

<script setup lang="ts">
  import { PropType } from 'vue'

  defineProps({
    type: {
      type: String as PropType<'text' | 'password' | 'email'>,
      default: 'text',
    },
    id: {
      type: String,
      required: true,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    readonly: {
      type: Boolean,
      default: false,
    },
    modelValue: {
      type: String,
      required: true,
    },
    labelName: {
      type: String,
      required: true,
    },
    labelPosition: {
      type: String as PropType<'top' | 'left'>,
      default: 'top',
    },
  })

  defineEmits(['update:modelValue'])
</script>

<style lang="scss" scoped>
  @import '@Bundles/UIBundle/Styles/UIInput.styles.scss';
</style>
