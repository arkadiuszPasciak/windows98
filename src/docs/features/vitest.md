# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
