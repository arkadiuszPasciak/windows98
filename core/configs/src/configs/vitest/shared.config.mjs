import { defineConfig } from 'vitest/config'

export default defineConfig({
    test: {
        environment: "jsdom",
        exclude: [
            "**/build/**",
            "**/dist/**",
            "**/node_modules/**",
            "**/@bundler/**",
            "**/@linter/**",
            "**/@test/**",
        ],
        include: [
            "**/*.unit.ts"
        ],
        globals: true,
    },
})
