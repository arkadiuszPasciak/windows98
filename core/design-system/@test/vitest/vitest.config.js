import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"

export default mergeConfig(
	sharedConfig,
	defineProject({/* here custom configs */ })
)
