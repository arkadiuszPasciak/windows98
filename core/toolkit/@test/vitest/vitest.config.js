import sharedConfig from "@windows98/configs/vitest/shared"
import { defineProject, mergeConfig } from "vitest/config"

export default mergeConfig(
	sharedConfig,
	defineProject({
		/* here custom configs */
	}),
)
