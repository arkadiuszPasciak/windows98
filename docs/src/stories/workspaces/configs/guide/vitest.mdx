import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

