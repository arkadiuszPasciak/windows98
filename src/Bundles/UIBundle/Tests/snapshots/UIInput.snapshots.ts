import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import UIInput from '@Bundles/UIBundle/Components/UIInput.vue'

const availableWrapper = mount(UIInput, {
  props: {
    variant: 'primary',
    type: 'email',
    id: 'main-email',
    modelValue: 'test@gmail.com',
    labelName: 'Email',
    labelPosition: 'left',
  },
})

const disabledWrapper = mount(UIInput, {
  props: {
    variant: 'primary',
    type: 'text',
    id: 'main-first-name',
    modelValue: 'John',
    labelName: 'First name',
    labelPosition: 'top',
    disabled: true,
  },
})

const readonlyWrapper = mount(UIInput, {
  props: {
    variant: 'primary',
    type: 'password',
    id: 'main-password',
    modelValue: '12345',
    labelName: 'Password',
    labelPosition: 'top',
    readonly: true,
  },
})

describe('[UIBundle]<Snapshots>(UIInput)', async () => {
  it('renders correctly component: availableWrapper like as email', () => {
    expect(availableWrapper.element).toMatchSnapshot()
  })

  it('renders correctly component: disabledWrapper like as text', () => {
    expect(disabledWrapper.element).toMatchSnapshot()
  })

  it('renders correctly component: readonlyWrapper like as password', () => {
    expect(readonlyWrapper.element).toMatchSnapshot()
  })
})
