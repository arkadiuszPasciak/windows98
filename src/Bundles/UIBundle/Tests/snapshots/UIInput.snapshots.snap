// Vitest Snapshot v1

exports[`[UIBundle]<Snapshots>(UIInput) > renders correctly component: availableWrapper like as email 1`] = `
<div
  class="UIInput label-position-left"
  data-v-e394574e=""
  variant="primary"
>
  <label
    class="label"
    data-v-e394574e=""
    for="main-email"
  >
    Email
  </label>
  <input
    class="input"
    data-v-e394574e=""
    id="main-email"
    type="email"
  />
</div>
`;

exports[`[UIBundle]<Snapshots>(UIInput) > renders correctly component: disabledWrapper like as text 1`] = `
<div
  class="UIInput label-position-top"
  data-v-e394574e=""
  variant="primary"
>
  <label
    class="label"
    data-v-e394574e=""
    for="main-first-name"
  >
    First name
  </label>
  <input
    class="input"
    data-v-e394574e=""
    disabled=""
    id="main-first-name"
    type="text"
  />
</div>
`;

exports[`[UIBundle]<Snapshots>(UIInput) > renders correctly component: readonlyWrapper like as password 1`] = `
<div
  class="UIInput label-position-top"
  data-v-e394574e=""
  variant="primary"
>
  <label
    class="label"
    data-v-e394574e=""
    for="main-password"
  >
    Password
  </label>
  <input
    class="input"
    data-v-e394574e=""
    id="main-password"
    readonly=""
    type="password"
  />
</div>
`;
